### ISSUE TITLE
Add codeboard.io

### ISSUE THREAD

**Issue by haches** [styfle/awesome-online-ide#25]
Codeboard.io is focusing on teaching / online courses.

Disclaimer: I'm one of the developers.

**Comment by styfle** [comment]
Hi @haches, thanks for the suggestion!

This looks like a Java Snippet runner similar to [Glot](https://glot.io/new/java) or [JDoodle](https://www.jdoodle.com/online-java-compiler) but you can't select the Java version or provide stdin.

Is there a particular feature that stands out with your IDE?

**Comment by haches** [comment]
Hi @styfle,

Codeboard currently supports the following languages: [C, C++, Eiffel, Haskell, Java, Python](https://codeboard.io/docs/project#languages)

Teachers can share template projects with students. The students then write and submit their code and it can be [automatically graded](https://codeboard.io/docs/submission#grading). Grading can be done via Unit tests or a custom test driver. The [teacher can inspect any submission](https://codeboard.io/docs/submission#inspect) a student makes.

Codeboard can be [easily integrated with Learning Management Systems](https://codeboard.io/docs/lti), making it possible to manage students and their grades outside of Codeboard.

You can have interactive programs (stdin, stdout): [here is an example, just compile and run it](https://codeboard.io/projects/8384).

Here's another [example of an coding exercise, using Java and JUnit tests](https://codeboard.io/projects/8450?view=21.1-22.0).

In summary: Codeboard is optimized for teaching programming, helping teachers to share, grade, and inspect code written by students, and integrate with other software used by schools, universities, and online courses.

Hope this makes it be a bit more clear. Otherwise, let me know :)

**Comment by styfle** [comment]
Yes that is much more clear, thanks!

I am debating if this is a Full IDE or just glorified snippets because there is no intellisense code-completion and the target market seems to be students.

But I like the tests feature which seems to be missing from a lot of other tools in the list.

Would you like to submit a PR?
